Change 26244 by nicholas[at]nicholas-saigo on 2005/12/02 17:15:27

        Address Hugo's comment on Dave's change (26240)

Change 26240 by davem[at]davem-splatty on 2005/12/01 16:40:29

        sprintf %NNN$ check for large values wrapping to negative

--- perl-5.8.3/sv.c.orig	2005-05-27 10:38:11 +0000
+++ perl-5.8.3/sv.c	2005-12-02 17:15:27 +0000
@@ -8655,9 +8655,15 @@ Perl_sv_vcatpvfn(pTHX_ SV *sv, const cha
 
 	if (vectorize)
 	    argsv = vecsv;
-	else if (!args)
-	    argsv = (efix ? efix <= svmax : svix < svmax) ?
-		    svargs[efix ? efix-1 : svix++] : &PL_sv_undef;
+	else if (!args) {
+	    if (efix) {
+		const I32 i = efix-1;
+		argsv = (i >= 0 && i < svmax) ? svargs[i] : &PL_sv_undef;
+	    } else {
+		argsv = (svix >= 0 && svix < svmax)
+		    ? svargs[svix++] : &PL_sv_undef;
+	    }
+	}
 
 	switch (c = *q++) {
 
